libcxx/vendor/llvm/default_assertion_handler.in

```diff
@@ -24,7 +24,13 @@
 #else
 
 #  if __has_builtin(__builtin_verbose_trap)
-#    define _LIBCPP_ASSERTION_HANDLER(message) __builtin_verbose_trap("libc++", message)
+// AppleClang shipped a slightly different version of __builtin_verbose_trap from the upstream
+// version before upstream Clang actually got the builtin.
+#    if defined(_LIBCPP_APPLE_CLANG_VER) && _LIBCPP_APPLE_CLANG_VER < 17000
```

Comment:
Does this mean AppleClang 17 will behave like upstream? If so let's add a TODO to remove this code once we AppleClang 16 will no longer be supported.